src/main/java/com/ldtteam/blockui/controls/ItemIconWithProperties.java

```diff
@@ -0,0 +1,165 @@
+ * {@value #NBT_GENERIC_KEY} - generic properties
+ * 
+ * @see ItemProperties
+ */
+@SuppressWarnings("deprecation")
+public class ItemIconWithProperties extends ItemIcon
+{
+    private static final String NBT_GENERIC_KEY = "_generic";
+    private static final String NBT_CURRENT_ITEM = "_item";
+
+    public static final String PARAM_PROPERTIES = "properties";
+
+    protected final Map<ResourceLocation, ItemPropertyFunction> genericPropertyOverrides = new HashMap<>();
+    protected final Map<Item, Map<ResourceLocation, ItemPropertyFunction>> itemPropertyOverrides = new HashMap<>();
+    private Map<ResourceLocation, ItemPropertyFunction> currentItemOverrides = Collections.emptyMap();
+
+    public ItemIconWithProperties()
+    {
+        super();
+    }
+
+    public ItemIconWithProperties(final PaneParams paneParams)
+    {
+        super(paneParams);
+
+        final String data = paneParams.getString(PARAM_PROPERTIES);
+        if (data != null && itemStack != null)
+        {
+            final CompoundTag tag;
+            try
+            {
+                tag = TagParser.parseTag(data);
+            }
+            catch (CommandSyntaxException e)
+            {
+                throw new RuntimeException(data, null);
+            }
+            tag.getAllKeys().forEach(itemKey -> {
+                if (tag.contains(itemKey, Tag.TAG_COMPOUND))
+                {
+                    final CompoundTag child = tag.getCompound(itemKey);
+                    final var itemOverrides = NBT_GENERIC_KEY.equals(itemKey) ? genericPropertyOverrides :
+                        itemPropertyOverrides.computeIfAbsent(NBT_CURRENT_ITEM.equals(itemKey) ? itemStack.getItem() :
+                            ForgeRegistries.ITEMS.getValue(ResourceLocation.tryParse(itemKey)), i -> new HashMap<>());
+
+                    child.getAllKeys().forEach(key -> {
+                        if (child.contains(key, Tag.TAG_ANY_NUMERIC))
+                        {
+                            final float value = child.getFloat(key); // intentionally ouf of lambda
+                            itemOverrides.put(ResourceLocation.tryParse(key), (itemStack, level, entity, seee) -> value);
+                        }
+                    });
+                }
+            });
+
+            onItemUpdate();
+        }
+    }
+
+    protected static String getNextItemKey(final String str, final int startIndex)
+    {
+        return str.substring(startIndex, str.indexOf('(', startIndex)).trim();
+    }
+
+    protected static String getNextItemOverrides(final String str, final int startIndex)
+    {
+        return str.substring(startIndex, str.indexOf(')', startIndex));
+    }
+
+    /**
+     * Short call for adding itemProperty to current item
+     */
+    public void addPropertyForCurrentItem(final ResourceLocation propertyKey, final ClampedItemPropertyFunction property)
```

Comment:
Any particular reason this uses `ClampedItemPropertyFunction`, and not just the interface `ItemPropertyFunction`?

Reply:
no particular reason, if you want to add already created property you can use getMap().put, this method is merely intended as syntax sugar for quick addition of new/local instances and therefore requires the guarded (clamped) version instead

Reply:
fixed